### DISCUSSION TITLE
Add custom variable in global config.yml for use in rule configs

### DISCUSSION THREAD

**Discussion by luminous706** [jertel/elastalert2#1120] [Q&A] [answered]
Hello, is it possible to add a custom variable to the `config.yml` file that can be used in a rule?

Here's what I tried but it doesn't work:

![image](https://user-images.githubusercontent.com/71604658/222256388-b084154b-e553-42d7-a3ca-d8679f63a281.png)

![image](https://user-images.githubusercontent.com/71604658/222256601-8e5872e3-9864-4b4a-8744-c69cbcddfd57.png)

**Comment by jertel** [comment, marked as answer]
Your command line args can contain references to match fields, as [documented](https://elastalert2.readthedocs.io/en/latest/ruletypes.html#command). However, I don't believe that will have access to config key/values. However, you have a couple of options:

1. You can write a tiny [enhancement](https://elastalert2.readthedocs.io/en/latest/recipes/adding_enhancements.html) to add custom keys/values to the match dict. The enhancement will have access to the rule config, so any key/values in there can also be copied into the match dict.

2. You can use the alert body text and args, which do allow references to config or rule key/values, combined with the `pipe_alert_text` setting (see documentation referenced earlier) to _pipe_ this in the form of JSON into your command.

**Reply by luminous706** [reply]
Thanks for the tips! I decided to use the command and do a sed in a script instead, but would be nice if we had something for straightforward to define variables.
